src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetVersionMappingControllerTests.cs

```diff
@@ -578,15 +661,176 @@ public async Task UpdateMappingDoesNotExist_Returns400_AndRollsBackTransaction()
+                                            CandidateKey = null
+                                        }
+                                    },
+                                    {
+                                        "source-location-3-key",
+                                        new LocationOptionMapping
+                                        {
+                                            Source = new MappableLocationOption("Source location 3")
+                                            {
+                                                Code = "Source location 3 code"
+                                            },
+                                            Type = MappingType.None,
+                                            CandidateKey = null
+                                        }
+                                    }
+                                },
+                                Candidates = new Dictionary<string, MappableLocationOption>()
+                                {
+                                    {
+                                        "target-la-location-1-key",
+                                        new MappableLocationOption("Target location 1")
+                                        {
+                                            Code = "Target location 1 code"
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                FilterMappingPlan = new FilterMappingPlan()
+            };
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersionMappings.Add(mappings);
+            });
+
+            var client = BuildApp().CreateClient();
+
+            List<LocationMappingUpdateRequest> updates =
+            [
+                // This candidate exists.
+                new()
+                {
+                    Level = GeographicLevel.LocalAuthority,
+                    SourceKey = "source-la-location-1-key",
+                    Type = MappingType.ManualMapped,
+                    CandidateKey = "target-la-location-1-key"
+                },
+                // This candidate does not exist.
+                new()
+                {
+                    Level = GeographicLevel.LocalAuthority,
+                    SourceKey = "source-la-location-2-key",
+                    Type = MappingType.ManualMapped,
+                    CandidateKey = "target-la-location-2-key"
+                },
+                // This candidate does not exist.
+                new()
+                {
+                    Level = GeographicLevel.Country,
+                    SourceKey = "source-la-location-3-key",
+                    Type = MappingType.ManualMapped,
+                    CandidateKey = "target-la-location-1-key"
```

Comment:
I'm not super fussed, but at first I didn't realise that the reason this candidate is invalid is because it is not a candidate for the `Country` level, even though it is a candidate for the `LocalAuthority` level.

This is a different failure scenario to the candidate key `target-la-location-2-key` above, which doesn't exist at all.

I just wondered if we should break these out into 2 separate tests to make it clearer - and because they could potentially go down different code paths? But I don't feel strongly either way

Reply:
Nice! Yep, I prefer this compromise actually :)